mcqa/data/utils.py

```diff
@@ -1,139 +1,375 @@
+        logger.info("LOOKING AT {} dev".format(data_dir))
+        high = os.path.join(data_dir, "dev/high")
+        middle = os.path.join(data_dir, "dev/middle")
+        high = self._read_txt(high)
+        middle = self._read_txt(middle)
+        return self._create_examples(high + middle, "dev")
 
+    def get_test_examples(self, data_dir):
+        """See base class."""
+        logger.info("LOOKING AT {} test".format(data_dir))
+        high = os.path.join(data_dir, "test/high")
+        middle = os.path.join(data_dir, "test/middle")
+        high = self._read_txt(high)
+        middle = self._read_txt(middle)
+        return self._create_examples(high + middle, "test")
 
-def read_mcqa_examples(input_file, is_training):
-    """Read an input file and return the corresponding MCQAExample instances.
+    def get_labels(self):
+        """See base class."""
+        return ["0", "1", "2", "3"]
 
-    Arguments:
-        input_file {str} -- File containing the data.
-        is_training {bool} -- Whether this is a training data
-        (labels not None).
+    def _read_txt(self, input_dir):
+        lines = []
+        files = glob.glob(input_dir + "/*txt")
+        for file in tqdm.tqdm(files, desc="read files"):
+            with open(file, "r", encoding="utf-8") as fin:
+                data_raw = json.load(fin)
+                data_raw["race_id"] = file
+                lines.append(data_raw)
+        return lines
 
-    Returns:
-        [MCQAExample] -- Returns a list of instances of ``MCQAExample`` .
-    """
+    def _create_examples(self, lines, set_type):
+        """Creates examples for the training and dev sets."""
+        examples = []
+        for (_, data_raw) in enumerate(lines):
+            race_id = "%s-%s" % (set_type, data_raw["race_id"])
+            article = data_raw["article"]
+            for i in range(len(data_raw["answers"])):
+                truth = str(ord(data_raw["answers"][i]) - ord("A"))
+                question = data_raw["questions"][i]
+                options = data_raw["options"][i]
 
-    with open(input_file, 'r', encoding='utf-8') as in_file:
-        reader = csv.reader(in_file)
-        lines = []
-        for line in reader:
-            lines.append(line)
+                examples.append(
+                    InputExample(
+                        example_id=race_id,
+                        question=question,
+                        # this is not efficient but convenient
+                        contexts=[article, article, article, article],
+                        endings=[options[0], options[1],
+                                 options[2], options[3]],
+                        label=truth,
+                    )
+                )
+        return examples
+
+
+class SwagProcessor(DataProcessor):
+    """Processor for the SWAG data set."""
+
+    def get_train_examples(self, data_dir):
+        """See base class."""
+        logger.info("LOOKING AT {} train".format(data_dir))
```

Comment:
```python
logger.info(f"LOOKING AT {data_dir} train")
```

Reply:
It can be useful for backward compatibility. But, I guess since python 2 support has ended, we can assume that python 3 code should be sufficient.